src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -217,15 +238,74 @@ private async Task<(Uri url, CancellationToken hostExitToken)> StartIISExpressAs
             }
         }
 
-        private string ModifyANCMPathInConfig(string replaceFlag, string dllName, string serverConfig, string contentRoot)
+        private void PrepareConfig(string contentRoot, string dllRoot, int port)
+        {
+            // Config is required. If not present then fall back to one we cary with us.
+            if (string.IsNullOrWhiteSpace(DeploymentParameters.ServerConfigTemplateContent))
+            {
+                using (var stream = GetType().Assembly.GetManifestResourceStream("Microsoft.AspNetCore.Server.IntegrationTesting.Http.config"))
+                using (var reader = new StreamReader(stream))
+                {
+                    DeploymentParameters.ServerConfigTemplateContent = reader.ReadToEnd();
+                }
+            }
+
+            var serverConfig = DeploymentParameters.ServerConfigTemplateContent;
+
+            // Pass on the applicationhost.config to iis express. With this don't need to pass in the /path /port switches as they are in the applicationHost.config
+            // We take a copy of the original specified applicationHost.Config to prevent modifying the one in the repo.
+            serverConfig = ModifyANCMPathInConfig(replaceFlag: "[ANCMPath]", dllName: "aspnetcore.dll", serverConfig, dllRoot);
```

Comment:
Lots of string replaces. I don't know what the performance hit is, but it could be decent. Consider using xml replacement or a StringBuilder?

Reply:
Yeah, that could use some work. It's not a measurable impact right now compared to the 3 seconds it takes dotnet to start the process 😁. I'm going to punt this re-design to a future PR.